C4.puml

```diff
@@ -1161,49 +1161,136 @@ $elementSkin
 !global $propTableCaption = ""
 !global $propColCaption = "="
 
+!global $isFirstProp = 1
+!global $firstPropCol = 1
+!global $lastPropCol = 1
+
+!function $fillMissing($col, $colNext)
+  !if ($col == "" && $colNext != "")
+    !return " "
+  !endif
+  !return $col 
+!endfunction
+
+!function $updatePropColumns($colIdx)
+  !if ($isFirstProp == 1 && $colIdx > $firstPropCol)
+    !$firstPropCol = $colIdx
+  !endif    
+  !if ($isFirstProp == 0 && $colIdx > $lastPropCol)
+    !$lastPropCol = $colIdx
+  !endif    
+  !return "" 
```

Comment:
Is there a reason for using an empty return statement, instead of making the `!function` a [`!procedure`][1]?

[1]: https://plantuml.com/preprocessing#01deeaa12f04fb9d

Reply:
If I would use a procedure then it would create an addititional newline and then the format would not match anymore